scripts/pm/hf_from_gcs.py

```diff
@@ -0,0 +1,224 @@
+from google.cloud import storage
+import os
+import re
+import subprocess
+import tempfile
+import shutil
+from collections import defaultdict
+import argparse
+
+# List of GCS directories to check
+dirs = [
+    "gs://marin-eu-west4/checkpoints/llama-8b-tootsie-0.001-19ad63/hf/",
+    "gs://marin-us-central2/checkpoints/llama-8b-tootsie-phase2/hf/",
+    "gs://marin-us-central2/checkpoints/llama-8b-tootsie-phase3/hf/",
+    "gs://marin-us-central2/checkpoints/tootsie-8b-soft-raccoon-3/hf/",
+    "gs://marin-us-central2/checkpoints/llama-8b-tootsie-adept-phoenix/hf/",
+    "gs://marin-us-central2/checkpoints/tootsie-8b-sensible-starling/hf/",
+    "gs://marin-us-central1/checkpoints/tootsie-8b-deeper-starling/hf/",
+]
+
+
+def list_gcs_directories(gcs_path):
+    """List subdirectories by examining full blob paths."""
+    if not gcs_path.startswith("gs://"):
+        raise ValueError(f"Invalid GCS path: {gcs_path}")
+
+    path = gcs_path[5:]  # Remove "gs://"
+    bucket_name = path.split("/")[0]
+    prefix = "/".join(path.split("/")[1:])
+
+    print(f"\nChecking: {gcs_path}")
+
+    # Get the bucket
+    client = storage.Client()
+    bucket = client.bucket(bucket_name)
+
+    # List blobs with this prefix (without delimiter to get all)
+    blobs = bucket.list_blobs(prefix=prefix)
```

Comment:
i'd think there'd be a more efficient method for this since you basically want to do a ~binary search or search sorted on whatever b-tree they're using

Reply:
the filtering the step dirs. but it's not important